packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -167,30 +213,42 @@ exports.sourceNodes = async (
         })
       }
 
-      const entryNode = {
+      let entryNode = {
         id: entryItem.sys.id,
         parent: contentTypeItemId,
         children: [],
-        ...entryItemFields,
         internal: {
           type: `${makeTypeName(contentTypeItemId)}`,
-          content: JSON.stringify(entryItem),
           mediaType: `application/json`,
         },
       }
 
+      // Replace text fields with text nodes so we can process their markdown
+      // into HTML.
+      Object.keys(entryItemFields).forEach(entryItemFieldKey => {
+        if (entryItemFieldKey === `text`) {
+          entryItemFields[`${entryItemFieldKey}___NODE`] = createTextNode(
+            entryNode,
+            entryItemFields[entryItemFieldKey],
+            createNode
+          )
+
+          delete entryItemFields[entryItemFieldKey]
+        }
+      })
+
+      entryNode = { ...entryItemFields, ...entryNode }
+
       // Get content digest of node.
-      const contentDigest = crypto
-        .createHash(`md5`)
-        .update(JSON.stringify(entryNode))
-        .digest(`hex`)
+      const contentDigest = digest(stringify(entryNode))
 
       entryNode.internal.contentDigest = contentDigest
 
       return entryNode
     })
+
     // Create a node for each content type
-    const contentTypeItemStr = JSON.stringify(contentTypeItem)
+    const contentTypeItemStr = stringify(contentTypeItem)
 
     const contentTypeNode = {
```

Comment:
Yeah, I wondered about this too...

One use case perhaps is someone building an internal documentation site from their content model.

I'd lean towards keeping it I guess. I assume Contentful exposes it because there's sufficient demand?

@Khaledgarbaya could you provide background for what people generally use this data for?